pkg/sentry/socket/socket.go

```diff
@@ -531,3 +531,123 @@ func IfconfIoctl(ctx context.Context, io usermem.IO, addr usermem.Addr) error {
 
 	return err
 }
+
+// InterfaceIoctl implements interface requests; and it returns corresponding interface
+// for customzation.
+func InterfaceIoctl(ctx context.Context, io usermem.IO, arg int, ifr linux.IFReq) (linux.IFReq, inet.Interface, *syserr.Error) {
+	var (
+		iface inet.Interface
+		index int32
+		found bool
+	)
+
+	// Find the relevant device.
+	stack := inet.StackFromContext(ctx)
+	if stack == nil {
+		return linux.IFReq{}, inet.Interface{}, syserr.ErrNoDevice
+	}
+
+	// SIOCGIFNAME uses ifr.ifr_ifindex rather than ifr.ifr_name to
+	// identify a device.
+	if arg == syscall.SIOCGIFNAME {
+		// Gets the name of the interface given the interface index
+		// stored in ifr_ifindex.
+		index = int32(usermem.ByteOrder.Uint32(ifr.Data[:4]))
+		if iface, ok := stack.Interfaces()[index]; ok {
+			ifr.SetName(iface.Name)
+			return ifr, iface, nil
+		}
+		return linux.IFReq{}, inet.Interface{}, syserr.ErrNoDevice
+	}
+
+	// Find the relevant device.
+	for index, iface = range stack.Interfaces() {
+		if iface.Name == ifr.Name() {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return linux.IFReq{}, inet.Interface{}, syserr.ErrNoDevice
+	}
+
+	switch arg {
+	case syscall.SIOCGIFINDEX:
+		// Copy out the index to the data.
+		usermem.ByteOrder.PutUint32(ifr.Data[:], uint32(index))
+
+	case syscall.SIOCGIFHWADDR:
+		// Copy the hardware address out.
+		ifr.Data[0] = 6 // IEEE802.2 arp type.
```

Comment:
This seems to be incorrect and undoing the fix made earlier to return the correct DeviceType.

See: https://github.com/google/gvisor/pull/1757/commits/d5930a9d0db4efd5e6aa49f11ecccd2d0894362e#diff-fef731bcd560ea8939f59fea252b0518L2996

Reply:
Also seems to be missing https://github.com/google/gvisor/pull/1757/commits/d5930a9d0db4efd5e6aa49f11ecccd2d0894362e#diff-fef731bcd560ea8939f59fea252b0518L3075 which I added recently.

Reply:
I don't know why; I actually already merged that change. I pushed again; please check again.